doc/reference/trd-appid.md

```diff
@@ -0,0 +1,795 @@
+TBF Object therefore requires using an alternative
+`TbfHeaderV2Program` header (the Program Header), which specifics
+where footers begin.
+
+The Tock process loading algorithm version numbers when deciding the
+order to load processes in. Version numbers are stored in a TBF Object
+in the Version field of a TBF Program Header.
+
+
+5.1 Program Header
+-------------------------------
+
+The Program Header is similar to the Main Header, in that it specifies
+the offset of the entry function of the executable and memory
+parameters. It adds one field, `binary_end_offset`, which indicates
+the offset at which the Userspace Binary ends within the TBF
+object. The space between this offset and the end of the TBF object is
+reserved for footers.
+
+This is the format of a Program Header:
+
+```
+0             2             4             6             8
++-------------+-------------+---------------------------+
+| Type (9)    | Length (16) | init_fn_offset            |
++-------------+-------------+---------------------------+
+| protected_size            | min_ram_size              |
++---------------------------+---------------------------+
+| binary_end_offset         | version                   |
++---------------------------+---------------------------+
+```
+
+It is represented in the Tock kernel with this Rust structure:
+
+```rust
+pub struct TbfHeaderV2Program {
+    init_fn_offset: u32,
+    protected_size: u32,
+    minimum_ram_size: u32,
+    binary_end_offset: u32,
+    version: u32,
+}
+```
+
+A TBF object MUST NOT have more than one Program Header. If a TBF
+Object has both a Program Header and a Main Header, the kernel's
+policy decides which is used. For example, older kernels that do not
+understand a Program Header may use the Main Header, while newer
+kernels may choose the Program Header.
+
+5.2 Credentials Footer
+-------------------------------
+
+To support credentials, the Tock Binary Format has a
+`TbfFooterV2Credentials` TLV. This TLV is variable length and has two
+fields, a 32-bit value specifying the `format` of the credentials and
+a variable length `data` field. The `format` field defines the format
+and size of the `data` field. Each value of the `format` field except
+`Reserved` MUST have a fixed data size and format. This is the format
+of a Credentials Footer:
+
+```
+0             2             4                           8
++-------------+-------------+---------------------------+
+| Type (128)  | Length      | format                    |
++-------------+-------------+---------------------------+
+| data                      |
++-------------+--------...--+
+```
+
+It is represented in the Tock kernel with this structure:
+
+```rust
+pub struct TbfFooterV2Credentials {
+    format: TbfFooterV2CredentialsType,
+    data: &[u8],
+}
+```
+
+Currently supported values of `format` are:
+
+```rust
+pub enum TbfFooterV2CredentialsType {
+    Reserved = 0,
+    CleartextID = 1,
+    Rsa3072Key = 2,
```

Comment:
The credential types are far from finalized. Since we have 32 bits I think the expectation is we will continue to refine which are supported and how they are structured.

Reply:
I understand, but why not reserve a `Rsa2048Key`?